javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -53,13 +48,13 @@
         Objects.requireNonNull(classifier, "classifier is null");
         Objects.requireNonNull(mapper, "mapper is null");
 
-        final java.util.Map<C, Collection<T>> mutableResults = new java.util.HashMap<>();
+        final java.util.Map<C, Collection<T>> mutableResults = new java.util.LinkedHashMap<>();
```

Comment:
Thank you! I can't review this PR now (out of my console), but it seems it significantly shorter than prev